### PR TITLE
fix: query param pattern

### PR DESCRIPTION
**Pull request by orensolo**
This pull request fixes issue https://github.com/getkin/kin-openapi/issues/641

### CODE REVIEW

File: openapi3filter/issue641_test.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  get:
+    description: Returns a list of stuff
+    parameters:
+    - description: test object
+      explode: false
+      in: query
+      name: test
+      required: false
+      schema:
+         allOf:
```
[the remark this answers is not in the document]
Comment (orensolo) [reply openapi3filter/issue641_test.go]: fixed.

File: openapi3filter/issue641_test.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+			route, pathParams, err := router.FindRoute(httpReq)
+			require.NoError(t, err)
+
+			requestValidationInput := &openapi3filter.RequestValidationInput{
+				Request:    httpReq,
+				PathParams: pathParams,
+				Route:      route,
+			}
+			err = openapi3filter.ValidateRequest(ctx, requestValidationInput)
+			require.Equal(t, testcase.isErr, err != nil)
```
[the remark this answers is not in the document]
Comment (orensolo) [reply openapi3filter/issue641_test.go]: fixed.  I use  "Contains"

File: openapi3filter/req_resp_decoder.go [hunk 3]
```diff
@@ -256,6 +256,20 @@ func decodeStyledParameter(param *openapi3.Parameter, input *RequestValidationIn
...
+// addSchemaTypeIfNeeded adds the schema type to the schema if
+// it can be concluded.
+// If the schema contain pattern without type and the parent has a string type,
+// then the type of the schema must be string as well
+func addSchemaTypeIfNeeded(parentType string, schema *openapi3.SchemaRef) {
+	if (schema != nil) && (parentType == "string") {
+
+		if (schema.Value.Pattern != "") &&
+			(schema.Value.Type == "") {
+			schema.Value.Type = "string"
```
[the remark this answers is not in the document]
Comment (orensolo) [reply openapi3filter/req_resp_decoder.go]: done.
Note that I limit the fix to urlValuesDecoder (possibly current or custom decoder may not need it)

**Comment by The-Murdock** [comment]
@fenollp it this PR ready for merge or anything else is needed?

**Comment by The-Murdock** [comment]
@fenollp  hi, Can you merge this fix or do you think something else is needed?

**Pull request merged**